src/java.desktop/macosx/classes/com/apple/laf/AquaButtonUI.java

```diff
@@ -307,8 +340,9 @@ public void paint(final Graphics g, final JComponent c) {
         // performs icon and text rect calculations
         final String text;
         final View v = (View)c.getClientProperty(BasicHTML.propertyKey);
-        if (v != null) {
-            // use zero insets for view since layout only handles text calculations
+        if (v != null && ((AbstractButton) c).getText().contains("<img ")) {
```

Comment:
You might want to test a combination of HTML image + icon added to JButton, to make sure the fix works correctly under different combinations.

Suggestion: Does having `((AbstractButton) c).getIcon() == null` make it more robust than checking for html tag patterns ?

Reply:
The icon isn't the HTML image like in the similar test `HtmlButtonImageTest`. This time the HTML is the text and the icon is simply a separate image. This is the cause of the issue. The importance will be if the HTML contents is actually an image or just HTML text.

Reply:
Nevermind, I see your point now that I've reread this. Checking for an icon achieves the same effect without having to check for an img pattern as long as one is adjusted. So, I think the icon check is more robust and cleaner. Thanks!

Reply:
@DamonGuy  Additionally, you might need to make sure the conditional and the fix works under different combinations especially when you have an html with <img> tag + icon added to the button.

Reply:
Yes, I tested with these combinations:
Icon with HTML image
<img width="274" alt="Screen Shot 2023-02-14 at 12 30 23 PM" src="https://user-images.githubusercontent.com/96267980/218855067-68186106-77df-4c2b-81d7-2cd24e28f4bb.png">

Icon with HTML text
<img width="270" alt="Screen Shot 2023-02-14 at 1 10 28 PM" src="https://user-images.githubusercontent.com/96267980/218863780-de2293ea-9da6-4db9-9201-4788b7567478.png">

And the `HtmlButtonImageTest` still passes as expected

Reply:
Looks good.

I might have missed it earlier, adding an instanceof check makes it safer before calling getIcon(). What do you think?
`if (v != null && c instanceof AbstractButton && ((AbstractButton) c).getIcon() == null)) {`

Reply:
Looks good.

I might have missed it earlier, adding an instanceof check makes it safer before calling getIcon(). What do you think?
`if (v != null && c instanceof AbstractButton && ((AbstractButton) c).getIcon() == null)){`

Reply:
Implemented your suggestion. I agree, it's more stable this way.